GliaWidgets/Sources/Extensions/UIButton+Extensions.swift

```diff
@@ -0,0 +1,24 @@
+import UIKit
+
+extension UIButton {
+    /// Sets the background color to use for the specified button state.
+    func setBackgroundColor(
+        color: UIColor,
+        forState: UIControl.State
+    ) {
+        let minimumSize: CGSize = CGSize(width: 1.0, height: 1.0)
+
+        UIGraphicsBeginImageContext(minimumSize)
+
+        if let context = UIGraphicsGetCurrentContext() {
+            context.setFillColor(color.cgColor)
+            context.fill(CGRect(origin: .zero, size: minimumSize))
+        }
+
+        let colorImage = UIGraphicsGetImageFromCurrentImageContext()
+        UIGraphicsEndImageContext()
+
+        self.clipsToBounds = true
```

Comment:
NIT: I personally like the use of `self` in this situation, but I believe it doesn't follow the rest of the codebase style.

Edit: I do see usages of this so it's up to you what you do.

Reply:
I think using `self` explicitly brings more clarity.